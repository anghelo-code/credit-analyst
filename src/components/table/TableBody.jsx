export const TableBody = ({ result }) => {
  
  return (
    <tbody>
      {
        result.map(item => (

        <tr key={item.id}>
          <td>{item.id}</td>
          <td>{item.amortizacion}</td>
          <td>{item.paInteres}</td>
          <td>{item.paMensual}</td>
          <td>{item.saldo}</td>
        </tr>

      ))}
    </tbody>
  )
}
