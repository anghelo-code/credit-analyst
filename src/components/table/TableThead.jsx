
export const TableThead = () => {
  return (
    <thead className="table-dark table-thead">
        <tr>
          <th scope="col">Periodo</th>
          <th scope="col">Amortización</th>
          <th scope="col">Pago de Interés</th>
          <th scope="col">Pago Mensual</th>
          <th scope="col">Saldo Capital</th>
        </tr>
    </thead> 
  )
}
