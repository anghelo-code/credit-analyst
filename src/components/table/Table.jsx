import { TableBody } from "./TableBody";
import { TableThead } from "./TableThead";
import PropTypes from "prop-types";

export const Table = ({ result }) => {
  return (
    <div className="row table-div">
      <div className="col-md-12">
          <table className="table table-bordered">

              <TableThead/>
              <TableBody result={ result }/>      
          
          </table>
      </div>
    </div>
  )
}

Table.propTypes = {
  result: PropTypes.array.isRequired
}
