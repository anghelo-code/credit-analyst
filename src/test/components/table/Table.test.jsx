import { render, screen } from "@testing-library/react";
import { Table } from "../../../components/table/Table";
import dataResult from '../../../data/datos.json';

describe('test of table body', () => { 

    test('should match the snapshot', () => { 
        const { container} = render( <Table result={ dataResult['datos'] } /> );
        expect(container).toMatchSnapshot();
    })

})