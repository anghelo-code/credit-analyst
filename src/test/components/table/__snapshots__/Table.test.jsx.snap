// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of table body should match the snapshot 1`] = `
<div>
  <div
    class="row table-div"
  >
    <div
      class="col-md-12"
    >
      <table
        class="table table-bordered"
      >
        <thead
          class="table-dark table-thead"
        >
          <tr>
            <th
              scope="col"
            >
              Periodo
            </th>
            <th
              scope="col"
            >
              Amortización
            </th>
            <th
              scope="col"
            >
              Pago de Interés
            </th>
            <th
              scope="col"
            >
              Pago Mensual
            </th>
            <th
              scope="col"
            >
              Saldo Capital
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              0
            </td>
            <td />
            <td />
            <td />
            <td>
              20000
            </td>
          </tr>
          <tr>
            <td>
              1
            </td>
            <td>
              1256.508
            </td>
            <td>
              1000
            </td>
            <td>
              2256.508
            </td>
            <td>
              18743.49
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              1319.333
            </td>
            <td>
              937.175
            </td>
            <td>
              2256.508
            </td>
            <td>
              17424.16
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              1385.3
            </td>
            <td>
              871.208
            </td>
            <td>
              2256.508
            </td>
            <td>
              16038.86
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              1454.565
            </td>
            <td>
              801.943
            </td>
            <td>
              2256.508
            </td>
            <td>
              14584.3
            </td>
          </tr>
          <tr>
            <td>
              5
            </td>
            <td>
              1527.293
            </td>
            <td>
              729.215
            </td>
            <td>
              2256.508
            </td>
            <td>
              13057.01
            </td>
          </tr>
          <tr>
            <td>
              6
            </td>
            <td>
              1603.657
            </td>
            <td>
              652.851
            </td>
            <td>
              2256.508
            </td>
            <td>
              11453.35
            </td>
          </tr>
          <tr>
            <td>
              7
            </td>
            <td>
              1683.84
            </td>
            <td>
              572.668
            </td>
            <td>
              2256.508
            </td>
            <td>
              9769.51
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td>
              1768.032
            </td>
            <td>
              488.476
            </td>
            <td>
              2256.508
            </td>
            <td>
              8001.48
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              1856.434
            </td>
            <td>
              400.074
            </td>
            <td>
              2256.508
            </td>
            <td>
              6145.05
            </td>
          </tr>
          <tr>
            <td>
              10
            </td>
            <td>
              1949.255
            </td>
            <td>
              307.253
            </td>
            <td>
              2256.508
            </td>
            <td>
              4195.8
            </td>
          </tr>
          <tr>
            <td>
              11
            </td>
            <td>
              2046.718
            </td>
            <td>
              209.79
            </td>
            <td>
              2256.508
            </td>
            <td>
              2149.08
            </td>
          </tr>
          <tr>
            <td>
              12
            </td>
            <td>
              2149.054
            </td>
            <td>
              107.454
            </td>
            <td>
              2256.508
            </td>
            <td>
              0.03
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
